src/rust/ensogl/src/math/geometry/triangle.rs

```diff
@@ -0,0 +1,169 @@
+//! This module defines an computations related to triangle geometry.
+
+use crate::math::algebra::Acos;
+use crate::math::algebra::Cos;
+use crate::math::algebra::Field;
+use crate::math::algebra::Sin;
+use crate::math::algebra::Sqrt;
+
+use core::f32::consts::PI;
+
+
+
+/// Represents a triangle through its angles and side lengths. This struct is only meant to be a
+/// return type within this module and should not otherwise be instantiated.
+///
+///
+/// Triangle Schematic
+/// -------------------
+/// ```text
+///                        C
+///                       / \
+///                     /    \
+///                   /       \
+///                 /  angle_c \
+///    side_ca    /             \  side_bc
+///             /                \
+///           /                   \
+///         /                      \
+///       /                         \
+///     /  angle_a           angle_b \
+///   /------------------------------- B
```

Comment:
Its super cool now! One question - do we need the image in the docs so big now? I mean, if you see `side_ab` you know this is between corners `a` and `b` right? Big ascii-arts makes the generated docs harder to read. Not a big deal though.

Reply:
I've reduced the sizes some more, but I think it useful to have a visual overview of all the terms.